### PR TITLE
gitea: new, 1.22.1

### PR DESCRIPTION
**Pull request by xunpod**
Topic Description
-----------------

- gitea: new, 1.22.1

Package(s) Affected
-------------------

- gitea: 1.22.1

Security Update?
----------------

No

Build Order
-----------

```
#buildit gitea
```

Test Build(s) Done
------------------

**Primary Architectures**

- [x] AMD64 `amd64`
- [x] AArch64 `arm64`
- [x] LoongArch 64-bit `loongarch64`

**Secondary Architectures**

- [x] Loongson 3 `loongson3`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`
- [x] RISC-V 64-bit `riscv64`

### CODE REVIEW

**Comment by xunpod** [comment]
Test passed on amd64.
image

**Comment by xunpod** [comment]
Set custom-path and work-path  to /etc/gitea.

image

**Comment by xunpod** [comment]
- Set the default custom-path to /etc/gitea, the  default  work-path to /var/lib/gitea/, and  the default config-path to /etc/gitea/app.ini.
image

- When first run, gitea will request a config  and write it to config-path,  so we needn`t provide a defaule config.
- If need to use systemd to manage gitea, users should creat a custom server config based on their own gitea config and [the examples](https://github.com/go-gitea/gitea/blob/main/contrib/systemd/gitea.service) provided by gitea.

File: app-web/gitea/autobuild/build [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+TAGS="bindata sqlite sqlite_unlock_notify pam" make build
+ 
+abinfo "Installing gitea ..."
+install -Dvm755 "$SRCDIR"/gitea "$PKGDIR"/usr/bin/gitea
+
+abinfo "Creating work-dir and config-dir ..."
+mkdir -pv "$PKGDIR"/var/lib/gitea/{custom,data,log}
+chmod -R 750 -v "$PKGDIR"/var/lib/gitea/
+mkdir -pv "$PKGDIR"/etc/gitea
+chmod 770 -v "$PKGDIR"/etc/gitea
```
Comment (OriginCode) [inline app-web/gitea/autobuild/build]: You can use `install -dvm [PERM] [PATH]` to easily create directories and modify the permission of the directories. See `man install` for more information.

Comment (xunpod) [reply app-web/gitea/autobuild/build]: Thanks, modified.

**Review by eatradish — CHANGES_REQUESTED** [review]
```
saki@MagMilk [ ~ ] $ gitea
2024/08/18 16:15:03 ...s/setting/setting.go:97:InitCfgProvider() [F] Unable to init config provider from "/etc/gitea/app.ini": unable to check if "/etc/gitea/app.ini" is a file. Error: stat /etc/gitea/app.ini: permission denied
```

**Pull request closed without merging**